ang/civicase/case/actions/services/email-managers-case-action.service.js

```diff
@@ -20,20 +23,24 @@
         }
       });
 
-      var popupPath = {
-        path: 'civicrm/activity/email/add',
-        query: {
-          action: 'add',
-          reset: 1,
-          cid: _.uniq(managers).join(',')
+      if (managers.length > 0) {
+        var popupPath = {
+          path: 'civicrm/activity/email/add',
+          query: {
+            action: 'add',
+            reset: 1,
+            cid: _.uniq(managers).join(',')
+          }
+        };
+
+        if (cases.length === 1) {
+          popupPath.query.caseid = cases[0].id;
         }
-      };
 
-      if (cases.length === 1) {
-        popupPath.query.caseid = cases[0].id;
+        return popupPath;
+      } else {
+        CRM.alert('Please add a contact as a case manager', 'No case managers available', 'error');
```

Comment:
Can we try to move this error as close as possible to the top of the function? You can put the condition plus a `return` statement.

With this we know right away why the call would fail and I think we'll remove a nesting level.